src/devices/src/virtio/net/device.rs

```diff
@@ -214,13 +215,18 @@ impl Net {
         self.tap.if_name_as_str().to_string()
     }
 
+    /// Provides the MmdsNetworkStack of this net device.
+    pub fn mmds_ns(&self) -> Option<&MmdsNetworkStack> {
+        self.mmds_ns.as_ref()
+    }
+
     /// Configures the `MmdsNetworkStack` to allow device to forward MMDS requests.
     /// If the device already supports MMDS, updates the IPv4 address.
-    pub fn configure_mmds_network_stack(&mut self, ipv4_addr: Ipv4Addr) {
+    pub fn configure_mmds_network_stack(&mut self, ipv4_addr: Ipv4Addr, mmds: Arc<Mutex<Mmds>>) {
```

Comment:
The `mmds` parameter feels like it should be an option, since we don't actually use it for all possible paths. It seems to be used only when we don't have an MMDS ns, so we shouldn't have to provide it for the other case. Alternatively we could split this in 2 functions.

Note, this will make more sense if we ever decide to not have MMDS on by default, so I'm fine with it being like this as of right now.